<a href="https://colab.research.google.com/github/gladysbabs/Machine-Learning-Project-Classification-Problem/blob/main/Project_2_Part_1_(Core)_Gladys_Babirye.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Project 2


## Project Overview: To predict metabolic syndrome, yes or not based on common risk factors.
- **DATASET 1:** Metabolic Syndrome.

In [1]:
#imports
import pandas as pd
import numpy as np

In [2]:
df=pd.read_csv('/content/drive/MyDrive/CodingDojo/02-IntroML/Week07/Data/Metabolic Syndrome.csv')
df.head()

seqn  Age     Sex  Marital  Income   Race  WaistCirc   BMI  Albuminuria  \
0  62161   22    Male   Single  8200.0  White       81.0  23.3            0   
1  62164   44  Female  Married  4500.0  White       80.1  23.2            0   
2  62169   21    Male   Single   800.0  Asian       69.6  20.1            0   
3  62172   43  Female   Single  2000.0  Black      120.4  33.3            0   
4  62177   51    Male  Married     NaN  Asian       81.1  20.1            0   

   UrAlbCr  UricAcid  BloodGlucose  HDL  Triglycerides MetabolicSyndrome  
0     3.88       4.9            92   41             84         No MetSyn  
1     8.55       4.5            82   28             56         No MetSyn  
2     5.07       5.4           107   43             78         No MetSyn  
3     5.22       5.0           104   73            141         No MetSyn  
4     8.13       5.0            95   43            126         No MetSyn

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2401 entries, 0 to 2400
Data columns (total 15 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   seqn               2401 non-null   int64  
 1   Age                2401 non-null   int64  
 2   Sex                2401 non-null   object 
 3   Marital            2193 non-null   object 
 4   Income             2284 non-null   float64
 5   Race               2401 non-null   object 
 6   WaistCirc          2316 non-null   float64
 7   BMI                2375 non-null   float64
 8   Albuminuria        2401 non-null   int64  
 9   UrAlbCr            2401 non-null   float64
 10  UricAcid           2401 non-null   float64
 11  BloodGlucose       2401 non-null   int64  
 12  HDL                2401 non-null   int64  
 13  Triglycerides      2401 non-null   int64  
 14  MetabolicSyndrome  2401 non-null   object 
dtypes: float64(5), int64(6), object(4)
memory usage: 281.5+ KB


- Source of data
   - Metabolic Syndrome Prediction dataset from data.world

- Brief description of data
   - This is data with the following variables: abnormal waist circumference, triglycerides above 150, HDL cholesterol below 50 in women or 40 in men, history of hypertension and mildly elevated fasting blood sugar (100-125), Albuminuria levels, and other variables such as uric acid, race, income, marital status, bmi, and sex. All this data might contribute to a suitable predictive model which shall be tested.

- What is the target?
   - MetabolicSyndrome

- What does one row represent? (A person? A business? An event? A product?)
  - One row represents a person.

- Is this a classification or regression problem?
   - With the target variable being categorical, this is a classification problem.

- How many features does the data have?
  - 13 features
- How many rows are in the dataset?
  - 2401 rows are in this dataset.

- What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
   - Cleaning: There are missing values that need to be addressed. There might be some data inconsistencies,and maybe duplicates as well.
   - Exploring: Missing values need to be addressed appropriately to ensure accurate exploration and preparation of the data for modeling.
   - Modeling: Selecting the appropriate models seems to be a foreseeable challenge for this dataset.


## Project Overview: Predicting the customer churn rate within the Telecom industry.
 - **DATASET 2**: Dataset on telecom_customer_churn.


In [4]:
df_cust=pd.read_csv('/content/drive/MyDrive/CodingDojo/02-IntroML/Week07/Data/churn_Data.csv')
df_cust.head()

tenure  age  address  income  ed  employ  equip  callcard  wireless  churn
0      11   33        7     136   5       5      0         1         1      1
1      33   33       12      33   2       0      0         0         0      1
2      23   30        9      30   1       2      0         0         0      0
3      38   35        5      76   2      10      1         1         1      0
4       7   35       14      80   2      15      0         1         0      0

In [5]:
df_cust.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 10 columns):
 #   Column    Non-Null Count  Dtype
---  ------    --------------  -----
 0   tenure    200 non-null    int64
 1   age       200 non-null    int64
 2   address   200 non-null    int64
 3   income    200 non-null    int64
 4   ed        200 non-null    int64
 5   employ    200 non-null    int64
 6   equip     200 non-null    int64
 7   callcard  200 non-null    int64
 8   wireless  200 non-null    int64
 9   churn     200 non-null    int64
dtypes: int64(10)
memory usage: 15.8 KB


- Source of data
  - This is a kaggle dataset on telecom customer churn.

- Brief description of data
  - This is a dataset with the following features: tenure, age, income,equipment availability, number of years employed,callcard and wireless technology.

- What is the target?
  - churn

- What does one row represent? (A person? A business? An event? A product?)
  - One row represents a person.

- Is this a classification or regression problem?
  - This is a binary classification problem predicting if the customer will churn(class 1) or not(class 0).

- How many features does the data have?
  - 9 features.

- How many rows are in the dataset?
  - 200 rows.

- What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
  - Cleaning: There might be duplicated data or even some data inconsistencies that will need to be addressed.
  - Exploring: I do not necessarily see any challenges on this.
  - Modeling: Finding the appropriate model to predict the churn rate will be challenging while testing various models or tuning them to get the most appropriate one.